massdriver-application-azure-app-service/main.tf

```diff
@@ -95,9 +95,10 @@ resource "azurerm_linux_web_app" "main" {
     }
 
     application_stack {
-      # configuring the structure this way so that the image block in massdriver.yaml is identical across azure runtimes
-      docker_image     = "${var.image.registry}/${var.image.name}"
-      docker_image_tag = var.image.tag
+      docker_registry_url      = var.image.registry
```

Comment:
This is turning into an `enum` or `oneOf` in the application template to choose registry source, i.e. Azure Container Registry, DockerHub (Public), or DockerHub (Private).

Reply:
We shouldn't prompt for the registry anywhere. We can infer it from the image URL. If there is no path, its Dockerhub. All other registries have the path in the URL, which should only be ACR.

If we need to manipulate the URL or do stuff with it we should automate/encode it.

Reply:
I added this in because with the new version, it's a requirement:
![image](https://github.com/massdriver-cloud/terraform-modules/assets/8037512/cbcf819e-7353-446f-8a7e-025526a41647)

Reply:
It's really dumb of Azure to do this, and we shouldn't pass it on to the user.

This should be deterministic. ACR images have `azurecr.io` in them, right? Dockerhub for sure doesn't. Just check for that in a local.